File: picker/src/main/java/com/ds/magicpicker/PickerBuilder.java
package com.ds.magicpicker;

import android.content.Context;
import android.content.Intent;

/**
 * Created by deepak on 14/3/18.
 */

public class PickerBuilder {

    private Context mContext;
    private boolean isCamera;

    public PickerBuilder(Context context) {
        mContext = context;
    }

    public PickerBuilder openCamera(boolean open) {
        isCamera = open;
        return this;
    }

    public PickerBuilder openGallery(boolean open) {
        isCamera = !open;
        return this;
    }

    public Intent build() {
        Intent intent = new Intent(mContext, PickerActivity.class);
        if (isCamera) {
            intent.putExtra(ImageContants.OPTION_CAMERA, true);
        } else {
            intent.putExtra(ImageContants.OPTION_GALLERY, true);
        }
        return intent;
    }
}

File: picker/src/main/java/com/ds/magicpicker/PickerBuilder.kt
package com.ds.magicpicker

import android.content.Context
import android.content.Intent

/**
 * Created by deepaksingh on 25/05/19.
 */
class PickerBuilder(private val mContext: Context) {
    private var isCamera: Boolean = false

    fun openCamera(): PickerBuilder {
        isCamera = true
        return this
    }

    fun openGallery(): PickerBuilder {
        isCamera = false
        return this
    }

    fun build(): Intent {
        val intent = Intent(mContext, PickerActivity::class.java)
        if (isCamera) {
            intent.putExtra(CONST_OPTION_CAMERA, true)
        } else {
            intent.putExtra(CONST_OPTION_GALLERY, true)
        }
        return intent
    }
}